client/src/request.js

```diff
@@ -1,38 +1,63 @@
-const photos = (productId) => {
-  return fetch(`http://ec2-3-136-203-39.us-east-2.compute.amazonaws.com:4002/photos/id/${productId}`)
-  .then(res => res.json())
-  .then(productPhotos => {
-    return productPhotos;
-  })
-  .catch(error => {
+const fetchWithTimeout = async (resource, options) => {
+  const { timeout = 4000 } = options;
+
+  const controller = new AbortController();
+  const id = setTimeout(() => controller.abort(), timeout);
+
+  const response = await fetch(resource, {
+    ...options,
+    signal: controller.signal
+  });
+  clearTimeout(id);
+
+  return response;
+};
+
+const photos = async (productId) => {
+
+  try {
+    const response = await fetchWithTimeout(`http://ec2-3-136-203-39.us-east-2.compute.amazonaws.com:4002/photos/id/${productId}`, {
+      timeout: 3000
+    });
+    const productPhotosInfo = await response.json();
+    return productPhotosInfo;
+  } catch (error) {
     let productPhotos = ['https://via.placeholder.com/640x480?text=Unable+to+get+primary+product+image'];
     for (let i = 2; i <= 5; i++) {
       productPhotos.push(`https://via.placeholder.com/640x480?text=Unable+to+get+product+image+${i}`);
     };
-    return productPhotos;
-  });
+    let errorProductPhotos = {
+      primaryUrl: productPhotos[0],
+      productUrls: productPhotos,
+    }
+    return errorProductPhotos;
+  }
+
 };
 
-const productInfo = (productId) => {
-  return fetch((`http://ec2-18-217-85-161.us-east-2.compute.amazonaws.com:4004/description/${productId}`))
-  .then(res => res.json())
-  .then((response) => {
-    let rawItemInfo = response[0];
+const productInfo = async (productId) => {
+
+  try {
+    const response = await fetchWithTimeout(`http://ec2-18-217-85-161.us-east-2.compute.amazonaws.com:4004/description/${productId}`, {
+      timeout: 3000
+    });
+    const productInfo = await response.json();
+    let rawItemInfo = productInfo[0];
     let itemInfo = {
       name: rawItemInfo.itemName,
       color: rawItemInfo.itemColor,
       configuration: rawItemInfo.configuration[0]
     };
     return itemInfo;
-  })
-  .catch((error) => {
+  } catch (error) {
```

Comment:
I like how you are handling your errors for any network request that doesn't go through successfully.

Reply:
Thanks!